rfcs/0149-cache-key-rotation.md

```diff
@@ -0,0 +1,73 @@
+---
+feature: (fill me in with a unique ident, my_awesome_feature)
+start-date: (fill me in with today's date, YYYY-MM-DD)
+author: (name of the main author)
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Rotate cache.nixos.org signing key.
+
+If we tolerate that people using stuff older than *X* could get a validation failure by default, fully rotating the key in 2*X* time seems relatively straightforward.
+
+# Motivation
+[motivation]: #motivation
+
+Never rotating a key is bad security practice.
+The current one has been in use at least since 2015.
+
+# Detailed design
+[design]: #detailed-design
+
+Preliminary action plan:
+- generate a new key
+- make it trusted by default (nix+nixpkgs, perhaps with backports to some branches)
+- wait until enough people trust the new key (at least one year, probably)
+- switch to signing with the new key
```

Comment:
Can we start signing with _both_ keys as soon as it's generated?

Reply:
I had that under preliminary consideration, even with some text.  Well, let me really push that text now 🤷🏽

Reply:
For me the important milestone isn't signing with the new key but the ability to remove trust for the old key (by default).